--- origsrc/lablgtk-2.12.0/src/ml_gdk.c	2008-10-30 12:02:27.000000000 -0500
+++ src/lablgtk-2.12.0/src/ml_gdk.c	2009-07-21 10:01:39.526769800 -0500
@@ -256,7 +256,7 @@ ML_1 (gdk_window_clear, GdkWindow_val, U
 ML_0 (GDK_ROOT_PARENT, Val_GdkWindow)
 ML_1 (gdk_window_get_parent, GdkWindow_val, Val_GdkWindow)
 
-#if defined(_WIN32) || defined(__CYGWIN__) || defined(HAS_GTKQUARTZ)
+#if defined(_WIN32) || defined(HAS_GTKQUARTZ)
 CAMLprim value ml_GDK_WINDOW_XWINDOW(value v)
 {
  ml_raise_gdk ("Not available for Win32");
@@ -491,7 +491,7 @@ CAMLprim value copy_xdata (gint format, 
 CAMLprim value ml_gdk_property_get (value window, value property,
                            value length, value pdelete)
 {
-#if defined(_WIN32) || defined(__CYGWIN__) || defined(HAS_GTKQUARTZ)
+#if defined(_WIN32) || defined(HAS_GTKQUARTZ)
   return Val_unit; /* not supported */
 #else
     GdkAtom atype;
