--- origsrc/lablgtk-2.10.0/config.make.in	2007-06-08 08:38:55.000000000 -0500
+++ src/lablgtk-2.10.0/config.make.in	2008-01-03 06:59:45.839558400 -0600
@@ -31,7 +31,7 @@
 RANLIB=@RANLIB@
 XA=.a
 XB=
-XE=
+XE=@EXE@
 XO=.o
 XS=.so
 TOOLCHAIN=unix
--- origsrc/lablgtk-2.10.0/configure.in	2007-06-08 08:38:55.000000000 -0500
+++ src/lablgtk-2.10.0/configure.in	2008-01-03 06:59:45.855183400 -0600
@@ -28,7 +28,7 @@
 #   OCAMLLIB      the path to the ocaml standard library
 #   OCAMLVERSION  the ocaml version number
 #   OCAMLWIN32    "yes"/"no" depending on Sys.os_type = "Win32"
-#   EXE           ".exe" if OCAMLWIN32=yes, "" otherwise
+#   EXE           ".exe" if Sys.os_type = "Win32" or "Cygwin", "" otherwise
 
 # check for one particular file of the sources 
 # ADAPT THE FOLLOWING LINE TO YOUR SOURCES!
@@ -201,13 +201,20 @@
 echo "print_endline Sys.os_type ;;" > conftest.ml
 ac_ocaml_platform=$(ocaml conftest.ml)
 AC_MSG_RESULT($ac_ocaml_platform)
-if test $ac_ocaml_platform = Win32 ; then
+case $ac_ocaml_platform in
+  Win32)
     OCAMLWIN32=yes
     EXE=.exe
-else
+    ;;
+  Cygwin)
+    OCAMLWIN32=no
+    EXE=.exe
+    ;;
+  *)
     OCAMLWIN32=no
     EXE=
-fi
+    ;;
+esac
 
 # GTK 2 auto configuration
 
--- origsrc/lablgtk-2.10.0/src/Makefile	2007-09-25 05:56:49.000000000 -0500
+++ src/lablgtk-2.10.0/src/Makefile	2008-01-03 06:59:45.855183400 -0600
@@ -333,7 +333,7 @@
 
 liblablgtkgl2$(XA): $(GLCOBJS)
 	$(LIBRARIAN) -o lablgtkgl2 $(GLCOBJS) \
-	    $(GTKGLLIBS)
+	    -ldopt -L. -llablgtk2 $(GTKGLLIBS)
 lablgtkgl.cma: liblablgtkgl2$(XA) $(GLMLOBJS)
 	$(LIBRARIAN) -o lablgtkgl -oc lablgtkgl2 $(GLMLOBJS) \
 	    $(GTKGLLIBS)
@@ -363,7 +363,7 @@
 	  -cclib "$(GLADELIBS)"
 else
 liblablglade2$(XA): $(GLADECOBJS)
-	$(LIBRARIAN) -o lablglade2 $(GLADECOBJS) $(GLADELIBS)
+	$(LIBRARIAN) -o lablglade2 $(GLADECOBJS) -ldopt -L. -llablgtk2 $(GLADELIBS)
 lablglade.cma: liblablglade2$(XA) $(GLADEMLOBJS)
 	$(LIBRARIAN) -o lablglade -oc lablglade2 $(GLADEMLOBJS) $(GLADELIBS)
 lablglade.cmxa: liblablglade2$(XA) $(GLADEMLOBJS:.cmo=.cmx)
@@ -386,7 +386,7 @@
 	  -cclib "$(RSVGLIBS)"
 else
 liblablrsvg$(XA): $(RSVGCOBJS)
-	$(LIBRARIAN) -o lablrsvg $(RSVGCOBJS) $(RSVGLIBS)
+	$(LIBRARIAN) -o lablrsvg $(RSVGCOBJS) -ldopt -L. -llablgtk2 $(RSVGLIBS) $(GTKLIBS)
 lablrsvg.cma: liblablrsvg$(XA) $(RSVGMLOBJS)
 	$(LIBRARIAN) -o lablrsvg -oc lablrsvg $(RSVGMLOBJS) $(RSVGLIBS)
 lablrsvg.cmxa: liblablrsvg$(XA) $(RSVGMLOBJS:.cmo=.cmx)
@@ -395,7 +395,7 @@
 endif
 
 liblablgnomecanvas$(XA): $(GNOMECANVASCOBJS)
-	$(LIBRARIAN) -o lablgnomecanvas $(GNOMECANVASCOBJS) $(GNOMECANVASLIBS)
+	$(LIBRARIAN) -o lablgnomecanvas $(GNOMECANVASCOBJS) -ldopt -L. -llablgtk2 $(GNOMECANVASLIBS)
 lablgnomecanvas.cma: liblablgnomecanvas$(XA) $(GNOMECANVASMLOBJS)
 	$(LIBRARIAN) -o lablgnomecanvas $(GNOMECANVASMLOBJS) $(GNOMECANVASLIBS)
 lablgnomecanvas.cmxa: liblablgnomecanvas$(XA) $(GNOMECANVASMLOBJS:.cmo=.cmx)
@@ -404,7 +404,7 @@
 # gnoCanvas.cmo: COMPILER+= -i
 
 liblablgnomeui$(XA): $(GNOMEUICOBJS)
-	$(LIBRARIAN) -o lablgnomeui $(GNOMEUICOBJS) $(GNOMEUILIBS)
+	$(LIBRARIAN) -o lablgnomeui $(GNOMEUICOBJS) -ldopt -L. -llablgtk2 $(GNOMEUILIBS)
 lablgnomeui.cma: liblablgnomeui$(XA) $(GNOMEUIMLOBJS)
 	$(LIBRARIAN) -o lablgnomeui $(GNOMEUIMLOBJS) $(GNOMEUILIBS)
 lablgnomeui.cmxa: liblablgnomeui$(XA) $(GNOMEUIMLOBJS:.cmo=.cmx)
@@ -413,7 +413,7 @@
 # gnoDruid.cmo: COMPILER+= -i
 
 liblablpanel$(XA): $(PANELCOBJS)
-	$(LIBRARIAN) -o lablpanel $(PANELCOBJS) $(PANELLIBS)
+	$(LIBRARIAN) -o lablpanel $(PANELCOBJS) -ldopt -L. -llablgtk2 $(PANELLIBS)
 lablpanel.cma: liblablpanel$(XA) $(PANELMLOBJS)
 	$(LIBRARIAN) -o lablpanel $(PANELMLOBJS) $(PANELLIBS)
 lablpanel.cmxa: liblablpanel$(XA) $(PANELMLOBJS:.cmo=.cmx)
@@ -421,7 +421,7 @@
 	    $(PANELMLOBJS:.cmo=.cmx) $(PANELLIBS)
 
 liblablgtkspell$(XA): $(GTKSPELLCOBJS)
-	$(LIBRARIAN) -o lablgtkspell $(GTKSPELLCOBJS) $(GTKSPELLLIBS)
+	$(LIBRARIAN) -o lablgtkspell $(GTKSPELLCOBJS) -ldopt -L. -llablgtk2 $(GTKSPELLLIBS)
 lablgtkspell.cma: liblablgtkspell$(XA) $(GTKSPELLMLOBJS)
 	$(LIBRARIAN) -o lablgtkspell $(GTKSPELLMLOBJS) $(GTKSPELLLIBS)
 lablgtkspell.cmxa: liblablgtkspell$(XA) $(GTKSPELLMLOBJS:.cmo=.cmx)
@@ -429,7 +429,7 @@
 	    $(GTKSPELLMLOBJS:.cmo=.cmx) $(GTKSPELLLIBS)
 
 liblablgtksourceview$(XA): $(GTKSOURCEVIEWCOBJS)
-	$(LIBRARIAN) -o lablgtksourceview $(GTKSOURCEVIEWCOBJS) $(GTKSOURCEVIEWLIBS)
+	$(LIBRARIAN) -o lablgtksourceview $(GTKSOURCEVIEWCOBJS) -ldopt -L. -llablgtk2 $(GTKSOURCEVIEWLIBS)
 lablgtksourceview.cma: liblablgtksourceview$(XA) $(GTKSOURCEVIEWMLOBJS)
 	$(LIBRARIAN) -o lablgtksourceview $(GTKSOURCEVIEWMLOBJS) $(GTKSOURCEVIEWLIBS)
 lablgtksourceview.cmxa: liblablgtksourceview$(XA) $(GTKSOURCEVIEWMLOBJS:.cmo=.cmx)
--- origsrc/lablgtk-2.10.0/src/ml_gdk.c	2007-09-24 21:56:09.000000000 -0500
+++ src/lablgtk-2.10.0/src/ml_gdk.c	2008-01-03 06:59:45.870808400 -0600
@@ -253,7 +253,7 @@
 ML_0 (GDK_ROOT_PARENT, Val_GdkWindow)
 ML_1 (gdk_window_get_parent, GdkWindow_val, Val_GdkWindow)
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__MINGW32__)
 CAMLprim value ml_GDK_WINDOW_XWINDOW(value v)
 {
  ml_raise_gdk ("Not available for Win32");
@@ -488,7 +488,7 @@
 CAMLprim value ml_gdk_property_get (value window, value property,
                            value length, value pdelete)
 {
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__MINGW32__)
   return Val_unit; /* not supported */
 #else
     GdkAtom atype;
